rotkehlchen/chain/evm/transactions.py

```diff
@@ -264,43 +262,44 @@ def _query_and_save_internal_transactions_for_range(
                         ),
                         has_premium=True,  # ignore limiting here
                     )
-                    if len(result) == 0:  # parent transaction is not in the DB. Get it
-                        transaction = self.evm_inquirer.get_transaction_by_hash(internal_tx.parent_tx_hash)  # noqa: E501
-                        gevent.sleep(0)
+                if len(result) == 0:  # parent transaction is not in the DB. Get it
+                    transaction = self.evm_inquirer.get_transaction_by_hash(internal_tx.parent_tx_hash)  # noqa: E501
+                    with self.database.conn.write_ctx() as write_cursor:
                         dbevmtx.add_evm_transactions(
-                            write_cursor=cursor,
+                            write_cursor=write_cursor,
                             evm_transactions=[transaction],
                             relevant_address=address,
                         )
-                        timestamp = transaction.timestamp
-                    else:
-                        timestamp = result[0].timestamp
+                    timestamp = transaction.timestamp
+                else:
+                    timestamp = result[0].timestamp
 
+                with self.database.conn.write_ctx() as write_cursor:
                     dbevmtx.add_evm_internal_transactions(
-                        write_cursor=cursor,
+                        write_cursor=write_cursor,
                         transactions=[internal_tx],
                         relevant_address=None,  # no need to re-associate address
                     )
-
-                    if period.range_type == 'timestamps':
-                        assert location_string, 'should always be given for timestamps'
-                        log.debug(f'Internal {self.evm_inquirer.chain_name} transactions for {address} -> update range {period.from_value} - {timestamp}')  # noqa: E501
+                if period.range_type == 'timestamps':
+                    assert location_string, 'should always be given for timestamps'
```

Comment:
should we keep assets like this in the code?

Reply:
You mean asserts? Some times you can do type ignore, some times you can do asserts. The key point is that asserts dissapear in production.

Reply:
yes asserts. True, I forgot that it gets removed. Thanks